src/super_gradients/training/datasets/collate_fn.py

```diff
@@ -0,0 +1,421 @@
+from torch.utils.data.dataloader import default_collate
+import random
+from typing import List, Union, Tuple, Optional, Dict, Type, Callable, Iterable
+
+from abc import ABC
+
+import cv2
+import torch
+import numpy as np
+
+
+from data_gradients.dataset_adapters.base_adapter import BaseDatasetAdapter
+from data_gradients.dataset_adapters.detection_adapter import DetectionDatasetAdapter
+from data_gradients.dataset_adapters.segmentation_adapter import SegmentationDatasetAdapter
+from data_gradients.dataset_adapters.classification_adapter import ClassificationDatasetAdapter
+
+from super_gradients.common.registry.registry import register_collate_function
+from super_gradients.common.decorators.factory_decorator import resolve_param
+from super_gradients.common.factories.collate_functions_factory import CollateFunctionsFactory
+from super_gradients.training.utils.detection_utils import xyxy2cxcywh
+
+from super_gradients.common.environment.device_utils import device_config
+from super_gradients.common.environment.ddp_utils import get_local_rank
+from super_gradients.training.utils.distributed_training_utils import wait_for_the_master
+from super_gradients.training.datasets.segmentation_datasets.segmentation_dataset import SegmentationDataSet
+
+
+class DatasetItemsException(Exception):
```

Comment:
Let's move expection classes to some subpackage?  E.g `super_gradients.exceptions` or `super_gradients.common.exceptions` ? Of course those exception classes should not have imports of any other classes, but I feel having some standardized place for all exceptions is a good thing.

Reply:
I think the main advantage is when someone builds a new class/function and wants to raise an exception, it's easier to find whether or not such exception already exists.
I'll open a PR for that

Reply:
There was already a dataset_exception apparently. I moved the exception there (among other) in this PR
https://github.com/Deci-AI/super-gradients/pull/1470